tests/integration-tests/conftest.py

```diff
@@ -490,7 +490,9 @@ def _add_policy_for_pre_post_install(node_config, custom_option, request, region
         logging.info("{0} script is not an S3 URL".format(custom_option))
     else:
         additional_iam_policies = {"Policy": f"arn:{_get_arn_partition(region)}:iam::aws:policy/AmazonS3ReadOnlyAccess"}
-        if dict_has_nested_key(node_config, ("Iam", "InstanceRole")):
+        if dict_has_nested_key(node_config, ("Iam", "InstanceRole")) or dict_has_nested_key(
+            node_config, ("Iam", "InstanceProfile")
+        ):
             # AdditionalIamPolicies and InstanceRole can not co-exist
             logging.info(
                 f"InstanceRole is specified, skipping insertion of AdditionalIamPolicies: {additional_iam_policies}"
```

Comment:
minor: the message should be: InstanceRole or InstanceProfile is specified...

Reply:
Changed